vyperlang/vyper
# chore(ci): add QOL improvements

### What I did

### How I did it

### How to verify it

### Commit message
```
chore(ci): add QOL improvements

* retry failing tests

tests which rely on timing (there is one crowdfunder test, and many
flaky tests) fail on the CI all the time presumably due to spiky CPU
slices. this uses the pytest-retriable package to auto retry failing
tests instead of having to baby-sit (and retry all tests!)

* use a matrix in GH actions instead of hardcoding python versions
* bump pytest-xdist
* add a bunch of runners
* add utility steps for GH status checks so we can have one check for an
  entire matrix of jobs
```

### Description for the changelog

### Cute Animal Picture

![Put a link to a cute animal picture inside the parenthesis-->]()
